terraform/eks/iam.tf

```diff
@@ -0,0 +1,186 @@
+#---
+# Provides a role for the cluster-autoscaler
+#
+# Note that the resource spec will result in a failure if we have
+# multiple ASGs. I would like to find a way to successfully pass
+# in a list.
+#---
+
+resource "aws_iam_role" "cluster_autoscaler" {
+  name = "cluster-autoscaler-${var.environment}-${var.region}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "AWS": "${module.eks.worker_iam_role_arn}"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "cluster_autoscaler" {
```

Comment:
I don't believe you need this here, the EKS Module actually creates it for you. https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/workers.tf#L193